### ISSUE TITLE
Promotion Rational number to Float32 fails

### ISSUE THREAD

**Issue by Chiil** [JuliaSIMD/LoopVectorization.jl#320]
In the code below, I follow the advice of the Julia language description to write my fraction as a `Rational` as this can be promoted to `Float32`, whereas `1/2` would be promoted to `Float64` and requires a cast to make it `Float32`. It does not work, however with the `@tturbo` macro with error: `ERROR: LoadError: promotion of types Rational{Int64} and VectorizationBase.VecUnroll{7, 16, Float32, VectorizationBase.Vec{16, Float32}} failed to change any arguments`

```
using LoopVectorization
using BenchmarkTools

function set_ghost_cells_bot_kernel!(
    a, a_bot, a_gradbot, dzh, ks)

    # fac = 0.5f0 # This works.
    fac = 1//2 # This does not work.

    @tturbo for j in 1:size(a, 2)
        for i in 1:size(a, 1)
            a[i, j, ks-1] = -a_gradbot[i, j]*dzh[ks] + a[i, j, ks]
            a_bot[i, j] = fac * (a[i, j, ks-1] + a[i, j, ks])
        end
    end
end

TF = Float32
# TF = Float64

a = rand(TF, 50, 50, 50)
a_bot = rand(TF, 50, 50)
a_gradbot = rand(TF, 50, 50)
dzh = rand(TF, 50)
ks = 2

@btime set_ghost_cells_bot_kernel!($a, $a_bot, $a_gradbot, $dzh, $ks)
```

**Comment by chriselrod** [comment]
Fixed by VectorizationBase 0.12.35.

**Comment by Chiil** [comment]
Thanks again. Thanks to this fix I have been able to implement the option for `Float32` or `Float64` in my code ([https://github.com/Chiil/MicroHH.jl](https://github.com/Chiil/MicroHH.jl)). It is amazing that with LoopVectorization I manage to reach performance comparable to an already highly optimized C++ code.
